drake/solvers/Constraint.h

```diff
@@ -182,17 +182,62 @@ class BoundingBoxConstraint : public LinearConstraint {
                          ub) {}
   virtual ~BoundingBoxConstraint() {}
 
-  virtual void eval(const Eigen::Ref<const Eigen::VectorXd>& x,
+  virtual void Eval(const Eigen::Ref<const Eigen::VectorXd>& x,
                     Eigen::VectorXd& y) const override {
-    y.resize(getNumConstraints());
+    y.resize(get_num_constraints());
     y = x;
   }
-  virtual void eval(const Eigen::Ref<const TaylorVecXd>& x,
+  virtual void Eval(const Eigen::Ref<const TaylorVecXd>& x,
                     TaylorVecXd& y) const override {
-    y.resize(getNumConstraints());
+    y.resize(get_num_constraints());
     y = x;
   }
 };
+
+
+/** LinearComplementarityConstraint
+ *
+ * Implements a constraint of the form:
+ *   Mx + q >= 0
+ *   x >= 0
+ *   x'(Mx + q) == 0
```

Comment:
This won't format correctly in Doxygen. You can use `<pre> untouched stuff </pre>` or doxgyen's verbatim command to slip unformatted stuff through.